src/protocol/v7.rs

```diff
@@ -1372,16 +1373,60 @@ fn is_other(value: &str) -> bool {
     value == "other"
 }
 
+static DEFAULT_FINGERPRINT: &'static [Cow<'static, str>] = &[Cow::Borrowed("{{ default }}")];
+
 #[cfg_attr(feature = "cargo-clippy", allow(ptr_arg))]
 fn is_default_fingerprint<'a>(fp: &Cow<'a, [Cow<'a, str>]>) -> bool {
     fp.len() == 1 && ((&fp)[0] == "{{ default }}" || (&fp)[0] == "{{default}}")
 }
 
+/// Recursively collects values into a fingerprint list.
+///
+/// - Booleans, strings and integers are directly converted to string
+/// - Floating point numbers are truncated and then converted to string
+/// - Arrays are transformed recursively with the above rules and appended to the list
+/// - Objects and null values are ignored at all levels
+fn collect_fingerprint<'a>(fingerprint: &mut Vec<Cow<'a, str>>, value: Value) {
+    match value {
+        Value::Null => (),
+        Value::Object(_) => (),
+        Value::Bool(b) => fingerprint.push(b.to_string().into()),
+        Value::String(s) => fingerprint.push(s.into()),
+        Value::Number(n) => {
+            if let Some(u) = n.as_u64() {
+                fingerprint.push(u.to_string().into());
+            } else if let Some(i) = n.as_i64() {
+                fingerprint.push(i.to_string().into());
+            } else if let Some(f) = n.as_f64() {
+                fingerprint.push(f.trunc().to_string().into());
```

Comment:
This will format into `inf`, `-inf` and `NaN` which is probably fine. If we want this behavior we need to make sure the python code does the same.

Reply:
I'm considering just not supporting float values that are not integers and to silently reject them. `10e100` is likely to format differently in different environments and unifying this would be hard. For instance Python formats them as `1e+101` whereas rust formats them as a whole number.